### ISSUE TITLE
Struct `new()` should warn in parameter default values

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#55021]
We are adding parameterless struct constructors (#51698) to the language. When a struct type adds a parameterless constructor, parameter default values which use `new()` will break (rather than quietly ignoring the parameterless constructor). We should discourage people from using `new()` in this context to reduce the frequency of such breaks.

```cs
// 1
static void func(S s = new(S)) { } // ok

public struct S { }
```

```cs
// 2
static void func(S s = new S()) { } // error

public struct S { public S() { } }
```

```cs
// 3
static void func(S s = default(S)) { } // ok

public struct S { public S() { } }
```

The suggestion here is to turn the *first* scenario into a warning.
